src/toxcore/dht/dht_node.rs

```diff
@@ -110,17 +155,35 @@ impl DhtNode {
                     }
                 },
                 None => {
-                    match self.saddr_v4 {
-                        Some(v4) => Some(SocketAddr::V4(v4)),
-                        None => None,
-                    }
+                    self.assoc4.saddr.map(Into::into)
                 },
             }
         } else {
-            match self.saddr_v4 {
-                Some(v4) => Some(SocketAddr::V4(v4)),
-                None => None,
-            }
+            self.assoc4.saddr.map(Into::into)
```

Comment:
if asso4 is empty we should try to check whether assoc6 is IPv6-mapped addr

Reply:
https://github.com/tox-rs/tox/blob/966afc3f921b6f9a765a6b819402ca9fcaaddc83/src/toxcore/dht/dht_node.rs#L88-L97

DhtNode always has at least one socket address.
So if IPv6 is None, IPv4 must be Some. Vice versa.
And If IPv6 is ipv4 mapped address, it is converted to IPv4 and stored in DhtNode.

Reply:
OK